### PR TITLE
Tl 53: Add Item to List and Render Items from its own reducer

### PR DESCRIPTION
**Pull request by peterevilla**
[Jira story](https://triplist.atlassian.net/secure/RapidBoard.jspa?rapidView=1&modal=detail&selectedIssue=TL-53)

### What does this PR do?
Creates Items reducer to manage list of items from list. Items are organized in a map of Ids.

Add item form is functional making a Post request to /items
### Screenshots

### CODE REVIEW

File: src/components/AddItemForm.tsx [hunk 1]
```diff
@@ -1,15 +1,38 @@
...
   const onchangeHandler = (e: any) => {
-    setNewItem(e.target.value);
+    setNewItem({ ...newItem, label: e.target.value });
   };
-
   const submitHandler = (e: any) => {
     e.preventDefault();
+    if (newItem.label !== '') {
+      createItem(newItem);
+      setTimeout(() => fetchList(listId), 200);
```
Comment (LuisRevillaM) [inline src/components/AddItemForm.tsx:33]: heheh I see what you did there

File: src/reducers/listByIdReducer.ts [hunk 2]
```diff
@@ -33,8 +33,9 @@ const listByIdReducer = (state = initialState, action: ListAction): ListState =>
     case FETCH_LIST_START:
       return { ...state, getListStatus: requestStatus.loading };
     case FETCH_LIST_SUCCESS:
+      const itemIds = action.payload.items.map((it: { id: any }) => it.id);
       return {
-        byId: { ...state.byId, [action.payload.id]: action.payload },
+        byId: { ...state.byId, [action.payload.id]: { ...action.payload, itemIds } },
         getListStatus: requestStatus.success,
       };
     case FETCH_LIST_ERROR:
```
Comment (LuisRevillaM) [inline src/reducers/listByIdReducer.ts:41]: Let's listen to `CREATE_ITEM_SUCCESS` here. 

from the payload we have to take the list id and push the item id on the `state.byId[listId].itemIds` array.

I will make the necessary change on the backend (we need the list ID on the payload).

File: src/components/AddItemForm.tsx (hunk 1 above)
Comment (peterevilla) [reply src/components/AddItemForm.tsx:33]: lol

File: src/reducers/listByIdReducer.ts (hunk 2 above)
Comment (peterevilla) [reply src/reducers/listByIdReducer.ts:41]: ok let me know!

Comment (LuisRevillaM) [reply src/reducers/listByIdReducer.ts:41]: done! 

new items include `listId`

**Pull request merged**